iree_tests/benchmarks/sdxl/benchmark_sdxl_rocm.py

```diff
@@ -181,6 +188,27 @@ def test_sdxl_rocm_benchmark(goldentime_rocm_e2e, goldentime_rocm_unet,
     job_summary_lines.append(mean_line)
     logging.getLogger().info(mean_line)
 
+    # unet compilation stats check
+    with open(f"{scheduled_unet_dir}/compilation_info.json", 'r') as file:
+        comp_stats = json.load(file)
+    unet_dispatch_count = int(comp_stats["stream-aggregate"]["execution"]["dispatch-count"])
+    assert unet_dispatch_count <= goldendispatch_rocm_unet, "SDXL scheduled unet dispatch count should not regress"
```

Comment:
It looks like these `assert`s will immediately fail the test. What we probably want is for the summary to always be generated and for _all_ errors to be reported, not just the first error. A way of doing that under pytest is using https://github.com/okken/pytest-check

1. add `pytest-check` to a `requirements.txt` file
2. add `from pytest_check import check`
3. modify these `assert` calls to use `check`, e.g.

    ```python
    check.less_equal(unet_dispatch_count, goldendispatch_rocm_unet, "SDXL scheduled unet dispatch count should not regress")
    ```
```

Reply:
Cool, updated

Reply:
The checks are not allowing the retries to occur. The retries are sometimes necessary to get the job passing. With it, we are able to be pretty strict with the values and have pretty solid testing so far. I moved the asserts to the end after the report is generated, so that we can stick with what's working for now (asserts/retries) and still get the report to show the status of every statistic.

Reply:
Ehh, this still only shows the first assert though.

```python
    assert benchmark_e2e_mean_time <= goldentime_rocm_e2e, "SDXL e2e benchmark time should not regress"
    assert benchmark_unet_mean_time <= goldentime_rocm_unet, "SDXL unet benchmark time should not regress"
    assert unet_dispatch_count == goldendispatch_rocm_unet, "SDXL scheduled unet dispatch count should not regress"
```

Do you have any theories on why the retries are helping? Are there other workloads running on the system? You could also try running with `--benchmark_min_warmup_time` (scaled to get enough iterations in).

I'd really like to avoid retries since they can complicate logs and interact poorly with non-flaky failures.

Otherwise I guess we could proceed with this and iterate on it later.

Reply:
I'm also concerned looking at those asserts that a developer would see "SDXL scheduled unet binary size should not get bigger" and then only update the `unet` binary size when the `vae decode` binary size might also need updating.

Reply:
Yeah, looks like --benchmark_min_warmup_time and increasing iterations to 10 works from my testing. One thing to note, if you rerun a job and the job recompiles the vmfb, it may slightly change the binary size. But, the timings always passed when I tested, so I will go with this for now :)